### PR TITLE
[mlir][vector][test] Split tests from vector-transfer-flatten.mlir

### PR DESCRIPTION
**Pull request by banach-space**
Move tests that exercise DropUnitDimFromElementwiseOps and
DropUnitDimsFromTransposeOp to a dedicated file.

While these patterns are collected under populateFlattenVectorTransferPatterns
(and are tested via -test-vector-transfer-flatten-patterns), they can actually
be tested without the xfer Ops, and hence the split.

Note, this is mostly just moving tests from one file to another. The only real
change is the removal of the following check-lines:

```mlir
//   CHECK-128B-NOT:   memref.collapse_shape
```

These were added specifically to check the "flattening" logic (which introduces
`memref.collapse_shape`). However, these tests were never meant to test that
logic (in fact, that's the reason I am moving them to a different file) and
hence are being removed as copy&paste errors.

I also removed the following TODO:

```mlir
/// TODO: Potential duplication with tests from:
///   * "vector-dropleadunitdim-transforms.mlir"
///   * "vector-transfer-drop-unit-dims-patterns.mlir"
```
I've checked what patterns are triggered in those test files and neither
DropUnitDimFromElementwiseOps nor DropUnitDimsFromTransposeOp does.

### CODE REVIEW

File: mlir/test/Dialect/Vector/vector-transfer-flatten.mlir [hunk 1]
```diff
@@ -538,186 +538,6 @@ func.func @transfer_write_non_contiguous_src(
...
-// CHECK-LABEL:   func.func @fold_unit_dim_add(
-// CHECK-SAME:      %[[VAL_0:.*]]: vector<8x1xi32>,
-// CHECK-SAME:      %[[VAL_1:.*]]: vector<1x8xi32>) -> vector<8xi32> {
-// CHECK:           %[[VAL_2:.*]] = vector.shape_cast %[[VAL_0]] : vector<8x1xi32> to vector<8xi32>
-// CHECK:           %[[VAL_3:.*]] = vector.shape_cast %[[VAL_1]] : vector<1x8xi32> to vector<8xi32>
-// CHECK:           %[[VAL_4:.*]] = arith.addi %[[VAL_2]], %[[VAL_3]] : vector<8xi32>
-// CHECK:           return %[[VAL_4]] : vector<8xi32>
-
-// CHECK-128B-LABEL: func @fold_unit_dim_add(
-//   CHECK-128B-NOT:   memref.collapse_shape
```
[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/test/Dialect/Vector/vector-transfer-flatten.mlir:601]: I don't believe those were meant to be here to begin with 😅 

Looking at the other tests, `CHECK-128B` was added specifically for flattening, which introduces `memref.collapse_shape`.  However, these tests have never tested flattening (that's the reason I am moving them). Specifically,  `//   CHECK-128B-NOT:   memref.collapse_shape` doesn't make much sense here, does it? Unless I am missing sth?

[the remark this answers is not in the document]
Comment (MacDue) [reply mlir/test/Dialect/Vector/vector-transfer-flatten.mlir:601]: Could you explain that in the summary? I didn't see the reason these were removed there, which is why I asked about it. :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (MacDue) [reply mlir/test/Dialect/Vector/vector-transfer-flatten.mlir:601]: Makes sense though (I can't see why `memref.collapse_shape` would appear here)

**Comment by banach-space** [comment]
> drop-unit-dims-with-shape-cast.mlir

Great suggestion, renamed 👍🏻

File: mlir/test/Dialect/Vector/vector-transfer-flatten.mlir (hunk 1 above)
[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/test/Dialect/Vector/vector-transfer-flatten.mlir:601]: Done 👍🏻

**Review by MacDue — APPROVED** [review]
LGTM

**Pull request merged**